cosmos/cache.py

```diff
@@ -400,3 +412,72 @@ def create_cache_profile(version: str, profile_content: str) -> Path:
     profile_yml_path = profile_yml_dir / DEFAULT_PROFILES_FILE_NAME
     profile_yml_path.write_text(profile_content)
     return profile_yml_path
+
+
+def is_cache_package_lockfile_enabled(project_dir: Path) -> bool:
+    package_lockfile = project_dir / PACKAGE_LOCKFILE_YML
+    return package_lockfile.is_file()
+
+
+def _get_sha1_hash(yaml_file: Path) -> str:
+    """Read package-lock.yml file and return sha1_hash"""
+    with open(yaml_file) as file:
+        yaml_content = file.read()
+    data = yaml.safe_load(yaml_content)
+    sha1_hash: str = data.get("sha1_hash", "")
+    return sha1_hash
+
+
+def _get_latest_cached_package_lockfile(project_dir: Path) -> Path | None:
```

Comment:
Would it make sense to use the method `_get_timestamp` that's being used to retrieve the newest partial parse file?
https://github.com/astronomer/astronomer-cosmos/blob/d5b3e9b7bc9063a7a44c0e583bf4e79f8f51fe09/cosmos/cache.py#L105

Reply:
When we deploy a dbt project, does it update the stats for only the file that has changed, or does it update the stats for all files? Example, if we make a change to a dbt model and deploy, does the stats update apply only to this modified model file, or to all files? If stats change for all files, then the current approach looks more reliable. WDYT?

Reply:
That's a good question, we need to check. There is a chance that as part of the Docker image build process, we're creating new timestamps. Could you please confirm in an Astro deployment so we know?

Reply:
I checked, and each time we deploy, the timestamp changes

Reply:
I'm resolving this conversion but please let me know if you have more input. thanks!